### PR TITLE
Display JUnit test results as they appear

### PR DESCRIPTION
**Pull request by basil**
Extracting a useful piece of #4447 into its own PR.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: Jenkinsfile
Comment (timja) [inline Jenkinsfile:65]: bit concerned that this will break checks api integration which is really nice:
https://github.com/jenkinsci/junit-plugin/issues/320

Comment (basil) [reply Jenkinsfile:65]: So what is the next step Tim? Close this PR?

Comment (timja) [reply Jenkinsfile:65]: I'll take a look at the junit plugin in between form modernisation work and see how hard it would be to just properly integrate it or patch real time junit

Comment (timja) [reply Jenkinsfile:65]: I think this PR would likely integrate it already: https://github.com/jenkinsci/junit-realtime-test-reporter-plugin/pull/12

Comment (timja) [reply Jenkinsfile:65]: I’m doing some testing on it and should have a release out by tomorrow so if we can just hold off till then that would be great (just want to make sure it’s viable)

File: Jenkinsfile
Comment (jglick) [inline Jenkinsfile:10]:
```suggestion
```

**Review by jglick — COMMENTED** [review]
Would be nice. Thanks @timja for helping to maintain this plugin.

File: Jenkinsfile
Comment (basil) [reply Jenkinsfile]: Done

**Comment by timja** [comment]
re-triggered good to go if it works

**Comment by basil** [comment]
Has the feedback from [this review](https://github.com/jenkinsci/jenkins/pull/6091#pullrequestreview-837543174) and [this review](https://github.com/jenkinsci/jenkins/pull/6091#pullrequestreview-838554089) been addressed? If so, is there a reason this PR has not yet been approved?

**Comment by timja** [comment]
> Has the feedback from this review and this review been addressed? If so, is there a reason this PR has not yet been approved?

just hadn't had a chance, 're-triggered good to go if it works' ...

**Review by timja — APPROVED** [review]

**Pull request merged**
